src/lib/datatypes/sol-buffer.c

```diff
@@ -30,6 +30,23 @@ nul_byte_size(const struct sol_buffer *buf)
     return (buf->flags & SOL_BUFFER_FLAGS_NO_NUL_BYTE) ? 0 : 1;
 }
 
+static void *
+secure_realloc(struct sol_buffer *buf, size_t new_size)
+{
+    char *new_data = NULL;
+
+    if (!new_size)
+        goto end;
+
+    new_data = malloc(new_size);
```

Comment:
Caller is checking that. I am freeing and clearing memory even on malloc errors.